app/error.tsx

```diff
@@ -1,6 +1,14 @@
 'use client';
 
-export default function Error({ reset }: { reset: () => void }) {
+import { useRouter } from 'next/navigation';
+
+export default function Error({ resetAction }: { resetAction?: () => void }) {
+  const router = useRouter();
+
+  const defaultReset = () => {
+    router.push('/');
+  };
```

Comment:
Since `resetAction` is optional, if there isn't a `resetAction`, the `Try again` button doesn't work. So I thought it would be better to set a `defaultReset` (e.g., navigate to the main page) to handle this.